downstairs/src/extent_inner_raw.rs

```diff
@@ -233,15 +233,21 @@ impl ExtentInner for RawInner {
                 (job_id.0, self.extent_number, n_contiguous_blocks as u64)
             });
 
-            // Perform the bulk read, then check against the expected number of
-            // bytes.  We could do more robust error handling here (e.g.
-            // retrying in a loop), but for now, simply bailing out seems wise.
-            let num_bytes = nix::sys::uio::pread(
-                self.file.as_fd(),
-                &mut buf,
-                first_req.offset.value as i64 * block_size as i64,
-            )
-            .map_err(|e| {
+            // SAFETY: the buffer has sufficient capacity, and this is a valid
+            // file descriptor.
+            let r = unsafe {
+                libc::pread(
+                    self.file.as_raw_fd(),
+                    buf.spare_capacity_mut().as_mut_ptr().cast(),
```

Comment:
extremely nitpicky, take it or leave it: personally, i try to avoid the use of un-turbofished `ptr::cast()`, as it obscures the type being cast to; i would generally either pronounce this as
```suggestion
                    buf.spare_capacity_mut().as_mut_ptr().cast::<libc::c_void>(),
```
or
```suggestion
                    buf.spare_capacity_mut().as_mut_ptr() as *mut libc::c_void,
```

but, feel free to ignore me!

Reply:
Sure, done in 028153cafc22c275282c946a6d0c131fcd41be76